### PR TITLE
feat!: node upgrade

### PR DESCRIPTION
**Pull request by vijayg10**
- Upgrade to node v16
- Update packages/linting/audit
- Update ci with dep/audit checks
- Installed pre-commit hooks

BREAKING CHANGE: Node version upgarde

Major version bump since this is a big upgrade.

### CODE REVIEW

File: .ncurc.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+reject: [
```
Comment (mdebarros) [inline .ncurc.yaml:1]:
```suggestion
## Add a TODO comment indicating the reason for each rejected dependency upgrade added to this list, and what should be done to resolve it (i.e. handle it through a story, etc).
reject: [
```

File: package.json [hunk 2]
```diff
@@ -42,20 +40,21 @@
...
-    "npm-audit-resolver": "2.3.1",
-    "npm-check-updates": "11.6.0",
-    "standard": "^16.0.3"
+    "@types/jest": "28.1.6",
+    "audit-ci": "^6.3.0",
+    "jest": "28.1.3",
+    "jest-junit": "14.0.0",
+    "npm-check-updates": "15.3.4",
+    "pre-commit": "^1.2.2",
+    "standard": "^17.0.0"
```
Comment (mdebarros) [inline package.json:57]: Please add `standard-version` `postchangelog` config as follows:

```suggestion
    "standard": "^17.0.0"
  },
  "standard-version": {
      "scripts": {
          "postchangelog": "replace '\\[mojaloop/#(\\d+)\\]\\(https://github.com/mojaloop/(.*)/issues/(\\d+)\\)' '[mojaloop/#$1](https://github.com/mojaloop/project/issues/$1)' CHANGELOG.md"
      }
```

NOTE:
- you will need to run `npm I -D replace` to ensure the dependency is met.
- please double check the indentation here

**Review by mdebarros — COMMENTED** [review]
Please some comments.

In addition:
- please add CODEOWNERS as well.
- Please add [.npmignore](https://github.com/mojaloop/central-ledger/blob/master/.npmignore) (feel free to add any additional directories/files to the ignore list)
- Please add the following to the end of your README.MD
  a. auditing-dependencies: https://github.com/mojaloop/central-ledger/blob/master/README.md#auditing-dependencies
  b. container-scans: https://github.com/mojaloop/central-ledger/blob/master/README.md#container-scans
  c. automated-releases: https://github.com/mojaloop/central-ledger/blob/master/README.md#automated-releases 


Take note that you should create a snapshot for testing on the OSS environment before any merges are done here, example PR where this is done: https://github.com/mojaloop/als-oracle-pathfinder/pull/72

File: README.md [hunk 3]
```diff
@@ -1 +1,6 @@
 # ml-testing-toolkit-shared-lib
+Library for shared code between Mojaloop testing-toolkit backend and frontend
+
+a. auditing-dependencies: https://github.com/mojaloop/central-ledger/blob/master/README.md#auditing-dependencies
```
Comment (mdebarros) [inline README.md:4]: What I meant by this is that you should copy and paste those sections into this document, not reference as you have done.

**Review by mdebarros — COMMENTED** [review]
Hey @vijayg10, see my comments.

Note that there are still a couple of issues that you did not address...like the Dockerfile suggestion that I made.

File: README.md (hunk 3 above)
Comment (vijayg10) [reply README.md]: Sorry for being so dumb :). Done

**Review by mdebarros — APPROVED** [review]
+1

**Pull request merged**